src/hooks/L1/useWriteDepositETH.ts

```diff
@@ -0,0 +1,64 @@
+import { useMutation } from '@tanstack/react-query'
+import { type Config, getChainId, getPublicClient, getWalletClient, type ResolvedRegister } from '@wagmi/core'
+import {
+  simulateDepositETH,
+  writeDepositETH,
+  type WriteDepositETHParameters as WriteDepositETHActionParameters,
+} from 'op-viem/actions'
+import { useConfig, type UseConfigReturnType } from 'wagmi'
+import type { UseWriteOPActionBaseParameters } from '../../types/UseWriteOPActionBaseParameters.js'
+import type { UseWriteOPActionBaseReturnType } from '../../types/UseWriteOPActionBaseReturnType.js'
+
+export type WriteDepositETHParameters = Omit<WriteDepositETHActionParameters, 'account'> & { chainId?: number }
+
+export type UseWriteDepositETHParameters<config extends Config = Config, context = unknown> =
+  UseWriteOPActionBaseParameters<WriteDepositETHParameters, config, context>
+
+export type UseWriteDepositETHReturnType<config extends Config = Config, context = unknown> =
+  & Omit<UseWriteOPActionBaseReturnType<WriteDepositETHParameters, config, context>, 'write' | 'writeAsync'>
+  & {
+    writeDepositETH: UseWriteOPActionBaseReturnType<WriteDepositETHParameters, config, context>['write']
+    writeDepositETHAsync: UseWriteOPActionBaseReturnType<WriteDepositETHParameters, config, context>['writeAsync']
+  }
+
+async function writeMutation(
+  config: UseConfigReturnType,
+  { chainId, ...params }: WriteDepositETHParameters,
+) {
+  const currentChainId = getChainId(config)
+  if (currentChainId !== chainId) throw new Error('chain mismatch')
+
+  const walletClient = await getWalletClient(config)
+  if (!walletClient) throw new Error('no account connected')
+  const publicClient = await getPublicClient(config)
+
+  await simulateDepositETH(publicClient, { ...params, account: walletClient.account.address, chain: undefined })
+  return writeDepositETH(walletClient, { ...params, account: walletClient.account.address, chain: undefined })
+}
+
+/**
+ * Deposits ETH to L2 using the OptimismPortal contract
+ * @param parameters - {@link UseWriteDepositETHParameters}
+ * @returns wagmi [useWriteContract return type](https://alpha.wagmi.sh/react/api/hooks/useWrtieContract#return-type). {@link UseWriteDepositETHReturnType}
+ */
+export function useWriteDepositETH<config extends Config = ResolvedRegister['config'], context = unknown>(
+  { mutation: mutationOverride }: UseWriteDepositETHParameters<config, context> = {},
+): UseWriteDepositETHReturnType<config, context> {
+  const config = useConfig()
+
+  const mutation = {
+    mutationFn(params: WriteDepositETHParameters) {
+      return writeMutation(config, params)
+    },
+    mutationKey: ['writeContract'],
```

Comment:
I honestly can't figure out what mutationKey actually does from the react-query docs, but do we want this to be something unique?

Reply:
at time of writing, these are all `writeContract` in wagmi too, so am going off of that. we can change if it's wrong / causes problems